### PR TITLE
Set CPU affinity for CVMFS processes

### PR DESCRIPTION
**Pull request by mharvey-jt**
It's occasionally useful to set the CPU affinity of the cvmfs2 processes. This patch add the configuration option `CVMFS_CPU_AFFINITY` which takes a comma-delimited list of cpu cores to bind to. Eg

CVMFS_CPU_AFFINITY=0,2

### CODE REVIEW

File: cvmfs/loader.cc [hunk 1]
```diff
@@ -736,6 +749,21 @@ int FuseMain(int argc, char *argv[]) {
...
+ if( options_manager->GetValue("CVMFS_CPU_AFFINITY", &parameter) ) {
+     cpu_set_t mask;
+     vector<int> cpus = splitToInts( parameter, ',' );
+     CPU_ZERO(&mask);
+     for( int i : cpus ) {
+        CPU_SET( i, &mask );
+     }
+
+     LogCvmfs(kLogCvmfs, kLogStdout, "Setting CPU Affinity to %s", parameter.c_str() );
+     int err = sched_setaffinity( 0, sizeof(mask), &mask );
```
Comment (jblomer) [inline cvmfs/loader.cc]: We'll need to ignore the request on macOS or use Mac's native API; I don't think `sched_setaffinity` exists there.

File: cvmfs/loader.cc [hunk 2]
```diff
@@ -736,6 +749,21 @@ int FuseMain(int argc, char *argv[]) {
     fuse_opt_add_arg(mount_options, "-osuid");
     LogCvmfs(kLogCvmfs, kLogStdout, "CernVM-FS: running with suid support");
   }
+
+ if( options_manager->GetValue("CVMFS_CPU_AFFINITY", &parameter) ) {
+     cpu_set_t mask;
+     vector<int> cpus = splitToInts( parameter, ',' );
```
Comment (jblomer) [inline cvmfs/loader.cc]: I think we should use `SplitString` and `String2UInt64` from the cvmfs string utils instead.

**Comment by jblomer** [comment]
Many thanks for the patch!  Do you have a recommendation how cvmfs pinning should be used?

**Comment by mharvey-jt** [comment]
This is useful in environments where we want to confine cvmfs to dedicated service cores, and not cause variable latency in other processes colocated on the host. This could also be done with a cgroup, but would be a more involved change.

File: cvmfs/loader.cc (hunk 1 above)
Comment (mharvey-jt) [reply cvmfs/loader.cc]: Made it a no-op on macos, as it's very unlikely to be a useful feature in that environment

File: cvmfs/loader.cc (hunk 2 above)
Comment (mharvey-jt) [reply cvmfs/loader.cc]: converted to use those

**Comment by jblomer** [comment]
Replaced by #3330

**Pull request closed without merging**